common/persistence/cassandra/mutable_state_store.go

```diff
@@ -747,13 +748,20 @@ func (d *MutableStateStore) ConflictResolveWorkflowExecution(
 		createRequestID := executionState.CreateRequestId
 		state := executionState.State
 		status := executionState.Status
-
-		executionStateDatablob, err := serialization.WorkflowExecutionStateToBlob(&persistencespb.WorkflowExecutionState{
-			RunId:           runID,
-			CreateRequestId: createRequestID,
-			State:           state,
-			Status:          status,
-		})
+		startTime := executionState.StartTime
+		details := executionState.Details
+
+		// TODO: should this just serialize the executionState variable?
```

Comment:
Wondering if we should just serialize `executionState` instead of building another object.

I already noticed that it was missing setting the `StartTime` below as well as the `LastUpdateVersioningTransition`.

For the `StartTime`, it's working because there's a handling for missing value since it's in transition phase from removing the field from `ExecutionInfo`.

For `LastUpdateVersioningTransition`, I don't know if it's supposed to copy from `executionState` set above in L741-746.

cc: @yycptt

Reply:
👍 Thanks! Yeah those two fields are missing. I didn't notice we were constructing a new state object here  when adding `LastUpdateVersioningTransition`. Guess the same thing happened for `StartTime`. 

Though at least for `LastUpdateVersioningTransition` I don't think there's any real impact. That field is only used via mutable state which is initialized by the concrete execution row that have the all the fields there. The blob constructed here (and I noticed the same thing for sql) is only for current execution row. 

cc @xwduan @hai719

Reply:
so we can just use the serialized blob in the `request`?

Reply:
Fixed! Now it's serializing the `executionState` var instead of copying.